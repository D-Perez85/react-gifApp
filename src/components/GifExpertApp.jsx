import React, { useState } from "react";
import AddCategory from "./AddCategory";
import GiftGrid from "./GiftGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);
  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((value) => 
        <GiftGrid category={value} key={value} />)}
      </ol>
    </>
  );
};
export default GifExpertApp;
