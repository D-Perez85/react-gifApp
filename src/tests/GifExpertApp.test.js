import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";
import "@testing-library/jest-dom";

/*
    1. Mostrar el componente GifGrid correctamente:  
    -shallow - wrapper - wrapper.toMatchSnapshot(); 
*/

describe("Pruebas en  <GifExpertApp/>", () => {
  let wrapper = shallow(<GifExpertApp />);
  test("Snapshot de GifExpertApp ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test("debe de mostrar una lista de categorias", () => {
  //   const categories = ["One Punch", "Dragon Ball"];
  //   const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

  //   expect(wrapper).toMatchSnapshot();
  //   expect(wrapper.find('GifGrid').length).toBe( categories.length);
  // });
});
