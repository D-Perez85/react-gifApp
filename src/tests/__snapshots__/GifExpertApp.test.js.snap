// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifExpertApp/> Snapshot de GifExpertApp  1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
